budget_control/models/base_budget_move.py

```diff
@@ -123,12 +123,16 @@ def _compute_commit(self):
         - Calc amount_commit from all budget_move_ids
         - Calc date_commit if not exists and on 1st budget_move_ids only or False
         """
+        period_id = self.env["budget.period"]._get_eligible_budget_period()
         for rec in self:
             debit = sum(rec.budget_move_ids.mapped("debit"))
             credit = sum(rec.budget_move_ids.mapped("credit"))
             rec.amount_commit = debit - credit
+            docline_period = rec[rec._budget_analytic_field].budget_period_id
             if rec.budget_move_ids:
```

Comment:
I still don't get this.

Normally, a docline will have budget_move_ids, so it always fall in here, isn't it?